main.cpp

```diff
@@ -222,43 +244,178 @@ IntType& IntType::divide(int i)
     return *this;
 }
 
+// ===== Point Constructors ===== //
+
+Point::Point(const FloatType& f) : x(f), y(f) {}
+Point::Point(const DoubleType& d) : x(static_cast<float>(d)), y(static_cast<float>(d)) {}
+Point::Point(const IntType& i) : x(static_cast<float>(i)), y(static_cast<float>(i)) {}
```

Comment:
You can make all of these forward to the constructor that takes 2 arguments: 
```
Point::Point(const T& t) : Point( static_cast<U>(t), static_cast<U>(t) ) { }
```
- [ ] revise to demonstrate knowledge of writing Forwarding Constructors